src/relative_duration/parse.rs

```diff
@@ -0,0 +1,191 @@
+    let (remainder, hours) = get_terminated(timespec, 'H')?;
+    let (remainder, mins) = get_terminated(remainder, 'M')?;
+    let (remainder, secs) = get_terminated(remainder, 'S')?;
+
+    if !remainder.is_empty() {
+        Err(format!(
+            "trailing characters: {} in timespec: {}",
+            remainder, timespec
+        ))
+    } else {
+        Ok((hours, mins, secs))
+    }
+}
+
+fn format_spec(nums: [i64; 3], chars: [char; 3]) -> String {
+    nums.iter()
+        .zip(chars)
+        .filter(|x| *x.0 != 0)
+        .map(|x| format!("{}{}", x.0, x.1))
+        .reduce(|acc, x| acc + &x)
+        .unwrap_or_else(|| "".to_string())
+}
+
+impl RelativeDuration {
+    /// Parses an [ISO 8601 duration string](https://en.wikipedia.org/wiki/ISO_8601#Durations) into
+    /// a [`RelativeDuration`] value.
+    ///
+    /// This supports only duration strings with integer values (i.e `"P1Y"` but not `"P0.5Y"` or
+    /// `"P0,5Y"`), as fractional values cannot be unambiguously represented as a [`RelativeDuration`]
+    ///
+    /// # Errors
+    ///
+    /// - Invalid duration string input
+    /// - Fractional values in duration string
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use chronoutil::RelativeDuration;
+    ///
+    /// assert_eq!(
+    ///     RelativeDuration::from_iso_8601("P1Y").unwrap(),
+    ///     RelativeDuration::years(1),
+    /// );
+    /// ```
+    pub fn from_iso_8601(input: &str) -> Result<RelativeDuration, String> {
+        let input = input
+            .strip_prefix('P')
+            .ok_or_else(|| "duration was not prefixed with P".to_string())?;
+
+        let (datespec, timespec) = input.split_once('T').unwrap_or((input, ""));
+
+        let (years, months, days) = parse_datespec(datespec)?;
+        let (hours, mins, secs) = parse_timespec(timespec)?;
+
+        Ok(RelativeDuration::months(years * 12 + months)
+            .with_duration(dhms_to_duration(days, hours, mins, secs)))
+    }
+
+    /// Formats a [`RelativeDuration`] value into an
+    /// [ISO 8601 duration string](https://en.wikipedia.org/wiki/ISO_8601#Durations).
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use chronoutil::RelativeDuration;
+    ///
+    /// assert_eq!(
+    ///     RelativeDuration::years(1).to_iso_8601(),
+    ///     "P1Y",
+    /// );
+    /// ```
+    pub fn to_iso_8601(&self) -> String {
+        let years = self.months as i64 / 12;
+        let months = self.months as i64 % 12;
+
+        let duration_seconds = self.duration.num_seconds();
+
+        // TODO: address this unwrap
+        let days = duration_seconds / (24 * 60 * 60);
+        let mut remaining_seconds = duration_seconds - (days * 24 * 60 * 60);
+
+        let hours = remaining_seconds / (60 * 60);
+        remaining_seconds -= hours * 60 * 60;
+
+        let minutes = remaining_seconds / 60;
+        remaining_seconds -= minutes * 60;
+
+        let seconds = remaining_seconds;
```

Comment:
I think this is clearer if you write it along the lines of
```rust
value = remainder / period
remainder = remainder % period
```

the compiler should also then be able to optimise this into a single divmod instruction (but you can benchmark to find out if it has any effect)

Reply:
I've made this change. Even if it doesn't lead to a compiler optimisation, I think your style is more readable.